src/Symfony/Component/HttpKernel/HttpCache/Psr6Store.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace Symfony\Component\HttpKernel\HttpCache;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class Psr6Store implements StoreInterface
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cacheBackend;
+
+    /**
+     * List of locks acquired by the current process.
+     *
+     * @var array
+     */
+    private $locks = array();
+
+    /**
+     * @param CacheItemPoolInterface $cacheBackend
+     */
+    public function __construct(CacheItemPoolInterface $cacheBackend)
+    {
+        $this->cacheBackend = $cacheBackend;
+    }
+
+    /**
+     * Locates a cached Response for the Request provided.
+     *
+     * @param Request $request A Request instance
+     *
+     * @return Response|null A Response instance, or null if no cache entry was found
+     */
+    public function lookup(Request $request)
+    {
+        // TODO: Implement lookup() method.
+    }
+
+    /**
+     * Writes a cache entry to the store for the given Request and Response.
+     *
+     * Existing entries are read and any that match the response are removed. This
+     * method calls write with the new list of cache entries.
+     *
+     * @param Request  $request  A Request instance
+     * @param Response $response A Response instance
+     *
+     * @return string The key under which the response is stored
+     */
+    public function write(Request $request, Response $response)
+    {
+        if (!$response->headers->has('X-Content-Digest')) {
+            $contentDigest = $this->generateContentDigest($response);
+            $item = $this->cacheBackend->getItem($contentDigest);
```

Comment:
Actually, I should create a CacheItem instead. No need to query the backend as we're going to overwrite the value anyway (if it's there).